go/arrow/array/concat.go

```diff
@@ -483,6 +587,18 @@ func concat(data []arrow.ArrayData, mem memory.Allocator) (arr arrow.ArrayData,
 		if err != nil {
 			return nil, err
 		}
+	case *arrow.ListViewType:
+		offsetType := arrow.PrimitiveTypes.Int32.(arrow.FixedWidthDataType)
```

Comment:
```go
		offsetType := arrow.PrimitiveTypes.Int32
		err := concatListView(data, offsetType, out, mem)
```

I get an error when I do this. Maybe the cast is not even valid (?) in the first place?

`cannot use offsetType (variable of type arrow.DataType) as arrow.FixedWidthDataType value in argument to concatListView: arrow.DataType does not implement arrow.FixedWidthDataType (missing method BitWidth)`

Reply:
Ah! darn me only declaring it as a `DataType` instead of actually as a `FixedWidthDataType` despite it implementing it. *sigh* Essentially you *could* do `&arrow.Int32Type{}` but that would be creating a new instance rather than using the existing one. 

I'd consider using the more specific interface type in the declared vars but I don't want to potentially break anyone who's already doing this type assertion....